salt/modules/cmdmod.py

```diff
@@ -2089,10 +2093,10 @@ def _cleanup_tempfile(path):
                                           saltenv,
                                           **kwargs)
         if not fn_:
-            if salt.utils.is_windows() and runas:
+            _cleanup_tempfile(path)
+            # If a temp working directory was created (Windows), let's remove that
+            if win_cwd:
                 _cleanup_tempfile(cwd)
```

Comment:
If I'm reading this right, doesn't `cwd` include `path`? And if so, do we really need two calls to `_cleanup_tempfile` in that case?

Reply:
It's a temporary directory created by the code above (line 2078) for the purposes of this function. After which it needs to be deleted

Reply:
If you don't pass `cwd` on a Windows box, this will create one under the `cachedir` with a temporary name. It will give the user permissions to that directory. The full path to the directory is in the `cwd` variable. The `path` variable is to a location inside the `cwd`, so if anything, we would only need to delete the `cwd` which would contain the `path`